### DISCUSSION TITLE
Can LGPL licensed library be used in commercial organizations?

### DISCUSSION THREAD

**Discussion by yvermabp** [LibrePDF/OpenPDF#1212] [General]
Can OpenPDF library be used in commercial organisation even though OpenPDF is fork of AGPL licensed library?

**Comment by ghost** [comment]
OpenPDF is not fork of any AGPL licensed code.

**Comment by mkl-public** [comment]
In more detail: iText changed to AGPL with version 5 and OpenPDF is a fork of version 2.1.7 (or the unpublished version 4.2.0 which in spite of the large gap is not much ahead of 2.1.7).
Also here people were cautious not to merge in any AGPL code.

**Comment by yvermabp** [comment]
In the [maven repository](https://mvnrepository.com/artifact/com.lowagie/itext) It is mentioned as AGPL license.
How can we be sure of this?

**Reply by ghost** [reply]
See detailed fork history, OpenPDF was forked from the last LGPL/MPL code of Itext. Originally Itext had this kind of licence. Later on licence of Itext was changed to agpl but this kind of code is not used in OpenPDF. I am not lawyer but for me is perfectly legal to use old code released under LGPL/MPL.

**Reply by mkl-public** [reply]
> In the [maven repository](https://mvnrepository.com/artifact/com.lowagie/itext) It is mentioned as AGPL license.

If you have the task of validating the licenses of the dependencies of your project, you should not believe what a third party (like mvnrepository) says but instead look for information by the original provider of that dependency.

You can find the take on this by Bruno Lowagie, the original creator and owner of iText [here on the iText knowledgebase](https://kb.itextpdf.com/it5kb/can-itext-2-1-7-itextsharp-4-1-6-or-earlier-be-use) or [here on stack overflow](https://stackoverflow.com/questions/25696851/can-itext-2-1-7-or-earlier-can-be-used-commercially) if you have the close/reopen votes privilege there. As you can read there, the arguments Bruno brought forth are not related to the license (which there is quoted to be MPL/LGPL) but to code sections whose origin is unclear or questionable.

That indeed can also be a hindrance to commercial use. But that's an issue you'll likely have in many other dependencies, too, without the developers of them mentioning that. And if you look around who all is using iText 2.1.7 to produce PDFs (there are certain fingerprints in the PDFs showing that they were created by iText 2.x or a recognizable fork thereof), you'll find quite a lot of the big and medium sized known companies among them...

> How can we be sure of this?

If you want to be really sure (or at least want the option to blame someone if an issue pops up), you'll need to buy a license of a library and make the seller by contract take any upcoming blame. Other than that you have to take a look at the details and decide for yourself.

----

PS: In this case you can actually read about the license change on mvnrepository, too. If you select the `4.2.2` version, you'll see this message:

![screen shot](https://github.com/user-attachments/assets/23106c8b-32e0-4bbe-b056-a68d1e29ea8f)

Nonetheless, in general don't simply believe what third parties web pages claim, search for credible documentation.

**Comment by yvermabp** [comment]
@kulatamicuda ok, I got it now, But if OpenPDF is fork of 2.1.7 How can we know that what are the features that OpenPDF doesn't support with latest version of itext like html conversion to pdf?

**Reply by ghost** [reply]
Sorry but I don’t understand your question. As I have written OpenPDF was created from latest LGPL/MPL code of Itext (the code before license switching in Itext occurred). This code is still and always will be LGPL/MPL. Later on this code was enhanced and updated by various volunteers including me (I am not longer active because of my work and family) under the same license. That’s all I can tell you about licensing. So far I know that OpenPDF is used in various projects including such projects which are currently being used in commercial applications as well.

**Comment by Lonzak** [comment]
I can understand that this is irritating - because the last license is shown at the top. This however doesn't reflect the actual facts / truth.

![grafik](https://github.com/user-attachments/assets/52565c46-1bf9-4ead-87b5-c9ba79312776)

But in the end this fits into the authors tactics to condemn people using this version of the library. He commercialized it and wanted to push everybody into using the paid version. Back in the days I had lots of discussions with him about this topic.

So in the end it was a good thing that openPDF was created to make a visible cut. And as you can see the license is correctly reflected:

![grafik](https://github.com/user-attachments/assets/5b4fa1bc-550f-4cb7-b0d4-f39c02194d3b)

**Comment by andreasrosdalw** [comment]
OpenPDF is licensed with Mozilla Public License Version 2.0 and GNU Lesser General Public License 2.1 as defined here:
https://github.com/LibrePDF/OpenPDF/blob/master/LICENSE.md

OpenPDF was forked from iText svn tag 4.2.0, which was licensed with Mozilla Public License Version 2.0 and GNU Lesser General Public License 2.1.

We really should encourage and help more open source projects migrate from iText to OpenPDF. 
iText 2.x has many security vulnerabilities, so upgrading to OpenPDF would be of benefit for other open source projects.
